### PR TITLE
viz: add check sub-command

### PR DESCRIPTION
**Pull request by Pothulapati**
This adds a new `viz check` cmd performing checks for the resources
in linkerd-viz extension. Checks include resource checks and
the health of resources, certs, etc

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: viz/cmd/check.go
Comment (adleong) [inline viz/cmd/check.go:87]: Is this anchor right?

File: viz/cmd/check.go
Comment (adleong) [inline viz/cmd/check.go:92]: Does this just check that the pods which exist in the namespace are running?

Should it also check that the correct pods exist in the first place?  Or perhaps that each of the expected deployments has a non-zero number of ready replicas?

Comment (Pothulapati) [reply viz/cmd/check.go]: Checking for the right pods to exist seemed to be the right thing.

 We can also add a another check for replica sets. WDYT?

File: pkg/healthcheck/healthcheck.go
Comment (alpeb) [inline pkg/healthcheck/healthcheck.go:1417]:
```suggestion
// CheckAPIService checks the status of the given API Service and returns an error if it's not running
```

File: viz/cmd/check.go
Comment (alpeb) [inline viz/cmd/check.go:263]:
```suggestion
	return nil, fmt.Errorf("could not find the linkerd-viz extension. It can be installed by running `linkerd viz install | kubectl apply -f -`")
```

File: viz/cmd/check.go
Comment (alpeb) [inline viz/cmd/check.go:201]: This is redundant with the "linkerd-viz Namespace exists"  check. That check should set a `hc.VizNamespace` as a side-effect for the other checks to use (I know it's a bit ugly to use a side-effect, but let's keep it so till we figure a better general framework).

File: viz/cmd/check.go
Comment (alpeb) [inline viz/cmd/check.go:122]:
```suggestion
		*healthcheck.NewChecker("tap API service is running").
```

**Review by alpeb — COMMENTED** [review]
Nice code reuse! :+1:

File: viz/cmd/check.go
Comment (Pothulapati) [reply viz/cmd/check.go:201]: That side-effect logic makes sense but adding `vizNamespace` under `hc` could be unnecessary. I can make the check update this global `vizNamespace` variable. wdyt?

Comment (alpeb) [reply viz/cmd/check.go:201]: I'm actually using such a `hc.vizNamespace` field to retrieve a viz api client in my public-api split work.

Comment (Pothulapati) [reply viz/cmd/check.go:201]: Hmmm. I'm okay with adding that `vizNamespace in `hc`. I can do that in this PR if there is no cocern. WDYT? @adleong

Comment (adleong) [reply viz/cmd/check.go]: I'm not dogmatic but here are my priorities in order:

1. The check command should be performant.  If re-creating a bunch of API clients make the command feel sluggish then we shouldn't do that.  If it has no noticeable performance impact, then let's not bend over backwards to be efficient
2. We should be consistent.  Let's follow the same patterns across the different check commands
3. Side-effects can sometimes be harder to reason about.  Let's avoid them if possible, but not at the expense of the above 2 priorities.

In this case, specifically, I think the redundancy of the namespace check is okay.  Since the first namespace check is fatal, we would expect that if we even get to this check, the namespace lookup should never fail.  (of course, we should still return the error if we somehow encounter it here, just to be defensive).

**Review by alpeb — APPROVED** [review]
Thanks @Pothulapati , this looks good to me :+1:

File: viz/cmd/check.go
Comment (Pothulapati) [reply viz/cmd/check.go]: Hmm, The current update i.e using `hc.nam|espace` with only a single namespace check that is fatal does seem to follow the priorities (especially the first two) and would also unblock Alejandros work.

**Comment by adleong** [comment]
There are a few things described in https://github.com/linkerd/linkerd2/issues/5325 which still need to be added here or in a follow-up PR:

* --proxy mode
* calling the SelfCheck API

File: viz/cmd/check.go
Comment (adleong) [inline viz/cmd/check.go:144]: tap-injector will need to be added here once it merges

**Review by adleong — APPROVED** [review]

**Pull request merged**
